tests/unit/test_equilibrium_single.py

```diff
@@ -0,0 +1,498 @@
+        assert len(coords) == len(funcs)
+        tmp = [f(c) for f, c in zip(funcs, coords)]
+        return reduce(lambda x, y: x * y, tmp)
+
+    return func
+
+
+def noisy_funcs(func, rel_sigma=0.02, abs_sigma=1e-3):
+
+    def noisy(*args):
+        y = func(*args)
+        max_std = 5
+        variance = max_std * rel_sigma * np.max(np.abs(y)) + \
+            max_std * abs_sigma
+        elements = np.random.random_sample(y.shape)
+        elements = (2.0 * elements) - 1.0
+        elements *= variance
+        return y + elements
+
+    return noisy
+
+
+def dropped_chnls(size, max_dropped=0.1):
+    return np.random.randint(
+        low=0,
+        high=size if size > 0 else 1,
+        size=int(max_dropped * size)
+    ).tolist()
+
+
+def data_arrays_from_coord(
+    data_type=(None, None),
+    coordinates=TrivialTransform(),
+    axes=(0.0, 0.0, 0.0),
+    data=separable_funcs(
+        smooth_funcs(max_val=1e3),
+        smooth_funcs(max_val=1e3),
+        smooth_funcs(max_val=1e3),
+    ),
+    rel_sigma=0.02,
+    abs_sigma=1e-3,
+    uncertainty=True,
+    max_dropped=0.1,
+    require_dropped=False,
+):
+    general_type = (data_type[0])  # To-do: error-checking
+    specific_type = (data_type[1])  # To-do: error-checking
+
+    x1, x2, t = axes
+    func = noisy_funcs(data, rel_sigma, abs_sigma) \
+        if rel_sigma or abs_sigma else data
+    coords = {}
+    dims = []
+    for n, c in [
+        ("t", t), (coordinates.x1_name, x1), (coordinates.x2_name, x2)
+    ]:
+        if isinstance(c, np.ndarray) and c.ndim > 0:
+            coords[n] = c.flatten()
+            dims.append(n)
+        elif isinstance(c, xr.DataArray) and c.ndim > 0:
+            coords[n] = c.values
+            dims.append(n)
+        elif not isinstance(coordinates, LinesOfSightTransform):
+            coords[n] = c if c is not None else 0.0
+    shape = tuple(len(coords[dim]) for dim in dims)
+    if isinstance(t, (np.ndarray, xr.DataArray)) and t.ndim > 0:
+        min_val = np.min(t)
+        width = np.abs(np.max(t) - min_val)
+        t_scaled = (t - min_val) / (width if width else 1.0)
+    else:
+        t_scaled = np.array([0.0])
+    if isinstance(x1, (np.ndarray, xr.DataArray)) and x1.ndim > 0:
+        min_val = np.min(x1)
+        width = np.abs(np.max(x1) - min_val)
+        x1_scaled = (x1 - min_val) / (width if width else 1.0)
+    else:
+        x1_scaled = np.array([0.0])
+    if isinstance(x2, (np.ndarray, xr.DataArray)) and x2.ndim > 0:
+        min_val = np.min(x2)
+        width = np.abs(np.max(x2) - min_val)
+        x2_scaled = (x2 - min_val) / (width if width else 1.0)
+    else:
+        x2_scaled = np.array([0.0])
+    tmp = func(x1_scaled, x2_scaled, t_scaled)
+    result = xr.DataArray(np.reshape(tmp, shape), coords, dims)
+
+    if isinstance(x1, np.ndarray):
+        flat_x1 = x1.flatten()
+    else:
+        flat_x1 = x1
+    dropped = (
+        [
+            flat_x1[i] for i in dropped_chnls(
```

Comment:
We probably don't need to worry about testing cases with dropped channels, either.